src/js/editor/editor.js

```diff
@@ -307,20 +299,27 @@ class Editor {
 
   handleDeletion(event) {
     event.preventDefault();
+    const key = Key.fromEvent(event),
+          direction = key.direction;
 
     let offsets = this.cursor.offsets;
     let currentMarker, currentOffset;
-    this.run((postEditor) => {
+
+    this.run(postEditor => {
       let results;
       if (this.cursor.hasSelection()) {
         results = postEditor.deleteRange(offsets);
       } else {
-        // FIXME: perhaps this should accept this.cursor.offsets?
-        results = postEditor.deleteCharAt(offsets.headMarker, offsets.headOffset-1);
+        const deletePosition = {
+          marker: offsets.headMarker,
+          offset: offsets.headOffset
+        };
+        results = postEditor.deleteFrom(deletePosition, direction);
```

Comment:
Hm, if we add a `deleteFrom` can it just be added on top of a `deleteCharAt` API? I imagine a generic character deletion API would be useful.

Or should the `deleteRange` just be used. Could be build both a `deleteCharAt` and `deleteFrom` on top of `deleteRange`? I found having all the logic for a given case flowing through one function was really useful for the linked list, ala `insertBefore` managing append, prepend, insertAfter etc.